client/src/components/NoteListView.js

```diff
@@ -42,22 +42,53 @@ const NoteContentListView = styled.p`
   margin: 0;
 `;
 
+const Highlighted = styled.span`
+  background-color: #a29cb38a;
+  transition: 0.3s;
+  border-radius: 5px;
+`;
+
+function getHighlightedText(text, searchQuery) {
+  if (text) {
+    // Split text on searchQuery term, include term itself into parts, ignore case
+    const parts = text.split(new RegExp(`(${searchQuery})`, 'gi'));
+    return (
+      <span>
+        {parts.map((part, index) =>
+          part.toLowerCase() === searchQuery.toLowerCase() ? (
+            <Highlighted key={index}>{part}</Highlighted>
+          ) : (
+            part
+          )
+        )}
+      </span>
+    );
+  }
+}
+
```

Comment:
This could/should be a Component because it returns JSX. 

```suggestion
function HighlightedText({ text, searchQuery }) {
  if (!text) {
    return null;
  }
 
   // Split text on searchQuery term, include term itself into parts, ignore case
  const parts = text.split(new RegExp(`(${searchQuery})`, 'gi'));
  
  return (
    <span>
      {parts.map((part, index) =>
        part.toLowerCase() === searchQuery.toLowerCase() ? (
          <Highlighted key={index}>{part}</Highlighted>
        ) : (
          part
        )
      )}
    </span>
  );
}

```